### PR TITLE
implement of basic account

### PR DESCRIPTION
**Pull request by baichuan3**
This PR is part of https://github.com/rooch-network/rooch/issues/16

implementation including:
- Account controlled by the private key
- Account controlled by ResourceAccount
- A global wall clock that stores the current Unix time in milliseconds
- A Draft implementation of block, transaction_validator and genesis module etc.

Among them, there are a few points that need to be explained
- SignerCapability can only be stored in other structs, and cannot be droped.
- The resource account can only by controlled by SignerCapability, there has no private key can control the resource account

### TODO
1, support rotate authentication key and manager rotate capability
2, complete transaction_validator and genesis module 
3, event register and emit

### CODE REVIEW

File: moveos/moveos-stdlib/rooch-framework/sources/account.move [hunk 1]
```diff
@@ -46,24 +99,213 @@ module rooch_framework::account{
...
+      } else {
+         create_account_unchecked(resource_addr)
+      };
+
+      // By default, only the SignerCapability should have control over the resource account and not the auth key.
+      // If the source account wants direct control via auth key, they would need to explicitly rotate the auth key
+      // of the resource account using the SignerCapability.
+      rotate_authentication_key_internal(&resource_signer, ZERO_AUTH_KEY);
+
+      assert!(!is_resource_account(resource_addr), error::invalid_state(EAccountIsAlreadyResourceAccount));
```
Comment (lshoo) [inline moveos/moveos-stdlib/rooch-framework/sources/account.move:209]: Is it better to move the `assert` statement before creating the signer? 
fail fast

File: moveos/moveos-stdlib/rooch-framework/sources/account.move [hunk 2]
```diff
@@ -46,24 +99,213 @@ module rooch_framework::account{
...
+   public(friend) fun rotate_authentication_key_internal(account: &signer, new_auth_key: vector<u8>) acquires Account {
+      let addr = signer::address_of(account);
+      assert!(exists_at(addr), error::not_found(EAccountNotExist));
+      assert!(
+         vector::length(&new_auth_key) == AUTHENTICATION_KEY_LENGTH,
+         error::invalid_argument(EMalformedAuthenticationKey)
+      );
+      let account_resource = borrow_global_mut<Account>(addr);
+      account_resource.authentication_key = new_auth_key;
+   }
```
Comment (lshoo) [inline moveos/moveos-stdlib/rooch-framework/sources/account.move:250]: Whether the `authentication_key` needs to be verified to be in a valid format？

File: moveos/moveos-stdlib/rooch-framework/sources/block.move [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    const EZeroEpochInterval: u64 = 1;
+
+
+    /// This can only be called during Genesis.
+    public(friend) fun initialize(account: &signer, epoch_interval_millisecs: u64) {
+        core_addresses::assert_rooch_genesis(account);
+        assert!(epoch_interval_millisecs > 0, error::invalid_argument(EZeroEpochInterval));
+
+        move_to<BlockResource>(
+            account,
```
Comment (lshoo) [inline moveos/moveos-stdlib/rooch-framework/sources/block.move:29]: Should we check the `BlockResource` exists in account before `move_to`?

File: moveos/moveos-stdlib/rooch-framework/sources/chain_id.move [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    const MAIN_CHAIN_ID: u8 = 1;
+    const DEV_CHAIN_ID: u8 = 181;
+    const TEST_CHAIN_ID: u8 = 182;
+
+    /// Only called during genesis.
+    /// Publish the chain ID under the genesis account
+    public fun initialize(account: &signer, id: u8) {
+        core_addresses::assert_rooch_genesis(account);
+        move_to(account, ChainId { id })
+    }
```
Comment (lshoo) [inline moveos/moveos-stdlib/rooch-framework/sources/chain_id.move:21]: Can developers call this function?
`public` -> `pubic(friend)`?

File: moveos/moveos-stdlib/rooch-framework/sources/account.move (hunk 2 above)
Comment (baichuan3) [reply moveos/moveos-stdlib/rooch-framework/sources/account.move:249]: `authentication_key` is generated by public keys via sha3-256 hash, and length is 32 bytes, has no other format

**Review by jolestar — APPROVED** [review]
Rush fast

**Pull request merged**